packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2782,6 +2782,60 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     }, /didInitAttrs called/);
   }
 
+  ['@test didInitAttrs is invoked before finishChains'](assert) {
+    expectDeprecation(/didInitAttrs called/);
+
+    let component;
```

Comment:
what do you think if we change the test to...

```js
this.registerComponent('foo-bar', {
  template: '{{foo}}-{{bar}}',

  ComponentClass: Component.extend({
    didInitAttrs({ attrs }) {
      this.set('bar', attrs.foo + 1);
    },

    barDidChange: ...
  })
});

assert.equal(fooDidChangeCount, 0);

this.render('{{foo-bar foo=foo}}', { foo: 1 });

assert.equal(fooDidChangeCount, 0);
this.assertText('1-2');

...

this.runTask(() => set(this.context, 'foo', 2);

...

```

Benefits (imo):

1. avoids the saving off `this` init dance
2. uses the attrs arg
3. (subjective) more meaningful/realistic